File: src/main/kotlin/Java/Hotel/Reserva.java
package Java.Hotel;

import java.util.Date;

class Reserva {
    private Habitacion habitacion;
    private Cliente cliente;
    private Date fechaInicio;
    private Date fechaFin;

    public Reserva(Habitacion habitacion, Cliente cliente, Date fechaInicio, Date fechaFin) {
        this.habitacion = habitacion;
        this.cliente = cliente;
        this.fechaInicio = fechaInicio;
        this.fechaFin = fechaFin;
    }

    public Habitacion getHabitacion() {
        return habitacion;
    }

    public Cliente getCliente() {
        return cliente;
    }

    public Date getFechaInicio() {
        return fechaInicio;
    }

    public Date getFechaFin() {
        return fechaFin;
    }

    @Override
    public String toString() {
        return "Reserva{" +
                "habitacion=" + habitacion +
                ", cliente=" + cliente +
                ", fechaInicio=" + fechaInicio +
                ", fechaFin=" + fechaFin +
                '}';
    }
}

File: src/main/kotlin/kotlinOPP/Hotel/Reserva.kt
package kotlinOPP.Hotel

import java.util.*

class Reserva(val habitacion: Habitacion, val cliente: Cliente, val fechaInicio: Date, val fechaFin: Date) {
    override fun toString(): String {
        return "Reserva(habitacion=$habitacion, cliente=$cliente, fechaInicio=$fechaInicio, fechaFin=$fechaFin)"
    }
}